### ISSUE TITLE
WER 0.46?

### ISSUE THREAD

**Issue by sonamsingh19** [cmusphinx/g2p-seq2seq#33]
I ran with 2 layers and 512 units but got nowhere close to reported?
Is this execution correct?
python -u  g2p.py --train ../../cmudict/cmudict.dict  --size 512
Preparing G2P data
Creating vocabularies in /tmp
Creating vocabulary /tmp/vocab.phoneme
Creating vocabulary /tmp/vocab.grapheme
Reading development and training data.
Creating 2 layers of 512 units.
Reading model parameters from /tmp/translate.ckpt-200
global step 400 learning rate 0.5000 step-time 2.78 perplexity 7.83
  eval: bucket 0 perplexity 4.88
  eval: bucket 1 perplexity 6.30
  eval: bucket 2 perplexity 12.34
global step 600 learning rate 0.5000 step-time 2.71 perplexity 4.34
  eval: bucket 0 perplexity 2.48
  eval: bucket 1 perplexity 2.96
  eval: bucket 2 perplexity 4.78
global step 800 learning rate 0.5000 step-time 2.63 perplexity 2.72
  eval: bucket 0 perplexity 1.75
  eval: bucket 1 perplexity 2.15
  eval: bucket 2 perplexity 3.45
global step 1000 learning rate 0.5000 step-time 2.56 perplexity 2.26
  eval: bucket 0 perplexity 1.65
  eval: bucket 1 perplexity 1.84
  eval: bucket 2 perplexity 3.17
global step 1200 learning rate 0.5000 step-time 2.68 perplexity 2.00
  eval: bucket 0 perplexity 1.29
  eval: bucket 1 perplexity 1.69
  eval: bucket 2 perplexity 2.57
global step 1400 learning rate 0.5000 step-time 2.86 perplexity 1.84
  eval: bucket 0 perplexity 1.48
  eval: bucket 1 perplexity 1.70
  eval: bucket 2 perplexity 2.15
global step 1600 learning rate 0.5000 step-time 3.40 perplexity 1.76
  eval: bucket 0 perplexity 1.65
  eval: bucket 1 perplexity 1.67
  eval: bucket 2 perplexity 2.18
global step 1800 learning rate 0.5000 step-time 3.65 perplexity 1.71
  eval: bucket 0 perplexity 1.42
  eval: bucket 1 perplexity 1.79
  eval: bucket 2 perplexity 2.04
global step 2000 learning rate 0.5000 step-time 2.68 perplexity 1.56
  eval: bucket 0 perplexity 1.30
  eval: bucket 1 perplexity 1.53
  eval: bucket 2 perplexity 1.83
global step 2200 learning rate 0.5000 step-time 3.33 perplexity 1.61
  eval: bucket 0 perplexity 1.50
  eval: bucket 1 perplexity 1.66
  eval: bucket 2 perplexity 1.70
global step 2400 learning rate 0.5000 step-time 3.01 perplexity 1.52
  eval: bucket 0 perplexity 1.29
  eval: bucket 1 perplexity 1.47
  eval: bucket 2 perplexity 1.79
global step 2600 learning rate 0.5000 step-time 3.09 perplexity 1.53
  eval: bucket 0 perplexity 1.34
  eval: bucket 1 perplexity 1.57
  eval: bucket 2 perplexity 1.90
global step 2800 learning rate 0.5000 step-time 2.92 perplexity 1.49
  eval: bucket 0 perplexity 1.35
  eval: bucket 1 perplexity 1.67
  eval: bucket 2 perplexity 1.85
global step 3000 learning rate 0.5000 step-time 2.82 perplexity 1.44
  eval: bucket 0 perplexity 1.39
  eval: bucket 1 perplexity 1.55
  eval: bucket 2 perplexity 1.81
global step 3200 learning rate 0.5000 step-time 2.68 perplexity 1.43
  eval: bucket 0 perplexity 1.49
  eval: bucket 1 perplexity 1.35
  eval: bucket 2 perplexity 1.87
global step 3400 learning rate 0.5000 step-time 2.90 perplexity 1.41
  eval: bucket 0 perplexity 1.35
  eval: bucket 1 perplexity 1.56
  eval: bucket 2 perplexity 1.73
global step 3600 learning rate 0.5000 step-time 2.79 perplexity 1.40
  eval: bucket 0 perplexity 1.27
  eval: bucket 1 perplexity 1.32
  eval: bucket 2 perplexity 1.59
global step 3800 learning rate 0.5000 step-time 2.87 perplexity 1.38
  eval: bucket 0 perplexity 1.52
  eval: bucket 1 perplexity 1.46
  eval: bucket 2 perplexity 1.52
global step 4000 learning rate 0.5000 step-time 2.74 perplexity 1.36
  eval: bucket 0 perplexity 1.49
  eval: bucket 1 perplexity 1.41
  eval: bucket 2 perplexity 1.83
global step 4200 learning rate 0.5000 step-time 2.80 perplexity 1.37
  eval: bucket 0 perplexity 1.23
  eval: bucket 1 perplexity 1.36
  eval: bucket 2 perplexity 1.58
global step 4400 learning rate 0.5000 step-time 2.94 perplexity 1.36
  eval: bucket 0 perplexity 1.58
  eval: bucket 1 perplexity 1.53
  eval: bucket 2 perplexity 1.73
global step 4600 learning rate 0.5000 step-time 3.16 perplexity 1.35
  eval: bucket 0 perplexity 1.25
  eval: bucket 1 perplexity 1.54
  eval: bucket 2 perplexity 1.58
global step 4800 learning rate 0.5000 step-time 2.74 perplexity 1.33
  eval: bucket 0 perplexity 1.44
  eval: bucket 1 perplexity 1.60
  eval: bucket 2 perplexity 1.72
global step 5000 learning rate 0.5000 step-time 2.77 perplexity 1.33
  eval: bucket 0 perplexity 1.36
  eval: bucket 1 perplexity 1.38
  eval: bucket 2 perplexity 1.60
global step 5200 learning rate 0.5000 step-time 2.97 perplexity 1.32
  eval: bucket 0 perplexity 1.29
  eval: bucket 1 perplexity 1.41
  eval: bucket 2 perplexity 1.66
global step 5400 learning rate 0.5000 step-time 2.77 perplexity 1.30
  eval: bucket 0 perplexity 1.31
  eval: bucket 1 perplexity 1.52
  eval: bucket 2 perplexity 1.45
global step 5600 learning rate 0.5000 step-time 2.80 perplexity 1.30
  eval: bucket 0 perplexity 1.31
  eval: bucket 1 perplexity 1.28
  eval: bucket 2 perplexity 1.75
global step 5800 learning rate 0.5000 step-time 2.64 perplexity 1.29
  eval: bucket 0 perplexity 1.42
  eval: bucket 1 perplexity 1.33
  eval: bucket 2 perplexity 1.41
global step 6000 learning rate 0.5000 step-time 2.76 perplexity 1.28
  eval: bucket 0 perplexity 1.26
  eval: bucket 1 perplexity 1.39
  eval: bucket 2 perplexity 1.48
global step 6200 learning rate 0.5000 step-time 2.55 perplexity 1.28
  eval: bucket 0 perplexity 1.37
  eval: bucket 1 perplexity 1.37
  eval: bucket 2 perplexity 1.67
global step 6400 learning rate 0.5000 step-time 2.68 perplexity 1.26
  eval: bucket 0 perplexity 1.23
  eval: bucket 1 perplexity 1.50
  eval: bucket 2 perplexity 1.44
global step 6600 learning rate 0.5000 step-time 2.98 perplexity 1.26
  eval: bucket 0 perplexity 1.12
  eval: bucket 1 perplexity 1.54
  eval: bucket 2 perplexity 1.47
global step 6800 learning rate 0.5000 step-time 2.87 perplexity 1.26
  eval: bucket 0 perplexity 1.22
  eval: bucket 1 perplexity 1.29
  eval: bucket 2 perplexity 1.56
global step 7000 learning rate 0.5000 step-time 2.81 perplexity 1.26
  eval: bucket 0 perplexity 1.22
  eval: bucket 1 perplexity 1.45
  eval: bucket 2 perplexity 1.54
global step 7200 learning rate 0.5000 step-time 2.76 perplexity 1.25
  eval: bucket 0 perplexity 1.35
  eval: bucket 1 perplexity 1.46
  eval: bucket 2 perplexity 1.40
global step 7400 learning rate 0.5000 step-time 3.06 perplexity 1.24
  eval: bucket 0 perplexity 1.18
  eval: bucket 1 perplexity 1.26
  eval: bucket 2 perplexity 1.48
global step 7600 learning rate 0.5000 step-time 3.15 perplexity 1.25
  eval: bucket 0 perplexity 1.47
  eval: bucket 1 perplexity 1.31
  eval: bucket 2 perplexity 1.50
global step 7800 learning rate 0.5000 step-time 3.13 perplexity 1.24
  eval: bucket 0 perplexity 1.50
  eval: bucket 1 perplexity 1.43
  eval: bucket 2 perplexity 1.46
global step 8000 learning rate 0.5000 step-time 2.76 perplexity 1.23
  eval: bucket 0 perplexity 1.39
  eval: bucket 1 perplexity 1.37
  eval: bucket 2 perplexity 1.47
global step 8200 learning rate 0.5000 step-time 2.64 perplexity 1.22
  eval: bucket 0 perplexity 1.30
  eval: bucket 1 perplexity 1.25
  eval: bucket 2 perplexity 1.59
global step 8400 learning rate 0.5000 step-time 2.38 perplexity 1.23
  eval: bucket 0 perplexity 1.42
  eval: bucket 1 perplexity 1.43
  eval: bucket 2 perplexity 1.45
global step 8600 learning rate 0.5000 step-time 2.53 perplexity 1.21
  eval: bucket 0 perplexity 1.42
  eval: bucket 1 perplexity 1.33
  eval: bucket 2 perplexity 1.39
global step 8800 learning rate 0.5000 step-time 2.58 perplexity 1.21
  eval: bucket 0 perplexity 1.21
  eval: bucket 1 perplexity 1.31
  eval: bucket 2 perplexity 1.50
global step 9000 learning rate 0.5000 step-time 2.88 perplexity 1.21
  eval: bucket 0 perplexity 1.36
  eval: bucket 1 perplexity 1.30
  eval: bucket 2 perplexity 1.57
global step 9200 learning rate 0.5000 step-time 3.03 perplexity 1.21
  eval: bucket 0 perplexity 1.47
  eval: bucket 1 perplexity 1.45
  eval: bucket 2 perplexity 1.38
global step 9400 learning rate 0.5000 step-time 2.77 perplexity 1.20
  eval: bucket 0 perplexity 1.39
  eval: bucket 1 perplexity 1.29
  eval: bucket 2 perplexity 1.55
global step 9600 learning rate 0.5000 step-time 2.86 perplexity 1.19
  eval: bucket 0 perplexity 1.53
  eval: bucket 1 perplexity 1.35
  eval: bucket 2 perplexity 1.46
global step 9800 learning rate 0.5000 step-time 2.87 perplexity 1.19
  eval: bucket 0 perplexity 1.43
  eval: bucket 1 perplexity 1.43
  eval: bucket 2 perplexity 1.80
global step 10000 learning rate 0.5000 step-time 2.74 perplexity 1.18
  eval: bucket 0 perplexity 1.36
  eval: bucket 1 perplexity 1.50
  eval: bucket 2 perplexity 1.45
Training process stopped.
Beginning calculation word error rate (WER) on test sample.
WER :  0.469490521327
Accuracy :  0.530509478673

**Comment by nurtas-m** [comment]
In order to get higher accuracy you need to set up max_steps parameter to 0 (no limit). By default it equals 10000 to speed up training. Maybe I should change the default setting 

By the way, where did you take this dict? There are several versions, actually

**Comment by sonamsingh19** [comment]
I'll rerun with  max_steps 0.

This dictionary is the one linked on the project github page:
http://github.com/cmusphinx/cmudict
